dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/tools/AbstractJspModC.java

```diff
@@ -285,6 +299,11 @@ private int _internalCompileApp(boolean doTraceCheck) throws JspCoreException {
                 try {
                     jdkLevel = Integer.parseInt(rawJdkLevel);
                 } catch(NumberFormatException e) {
+                        if (logger!=null) {
+                            if(doTraceCheck&&logger.isLoggable(Level.WARNING)){
+                                logger.logp(Level.WARNING,CLASS_NAME,"_internalCompileApp", "failed to parse jdkSourceLevel: " + rawJdkLevel, e);
+                            }
+                        }
```

Comment:
As far as I can tell, this class isn't used in OL, so I have no idea how to test changes to it.  AbstractJSPModC is used by JspBatchCompiler in WebSphere, but that doesn't exist in liberty.